aips/aip-10X.md

```diff
@@ -0,0 +1,120 @@
+---
+
+# AIP-90 - Add Firebase as a supported OIDC provider
+
+## Summary
+
+We are proposing adding Firebase as a supported OIDC provider for Aptos Keyless accounts.
+This would allow the instantiation of Keyless accounts using Firebase JWT credentials.
+This change requires Aptos validators to periodically refresh their views of Firebase's JWKs[^jwks] via JWK consensus[^aip-67].
+JWTs used to authenticate into Firebase apps are signed by the same set of keys across all Firebase applications.  Thus it makes sense to integrate the key set via JWK consensus.  Google uses the securetoken@system.gserviceaccount.com service account as the signer.
+
+The UWL of the well-known OpenId configuration is - https://securetoken.google.com/google/.well-known/openid-configuration
+
+## Impact and risks
+
+- In Firebase the `iss` and `aud` values are the same.  Since `iss` is public `aud` will be exposed.
+- Firebase allows for full custumizable authentication by the project developer.  This allows the developer to generate JWTs for users (scoped to their project via the `iss`) at will.
+- Since all Firebase apps use a common keyset, we are able to utilize JWK consensus as there are no scaling issues if the number of apps using Firebase grows larger, since the number of JWKS to register stays the same.  
+- Due to the above, we can utilize standard Keyless Accounts (non-Federated) but these accounts encounter have the same risks and considerations as Federated Keyless Accounts from a trust perspective.  However loss of access due to mismanagement of the JWK owner account is no longer an issue.
+- The well known configuration address is only assumed to be stable.  It will exist so long as there exists a Firebase project ID with the value 'Google'.  This is not publicized anywhere and its unclear of guarantees of its availability.  All Firebase apps have a well known discovery endpoint at `https://securetoken.google.com/<projectId>/.well-known/openid-configuration` which point to the same JWKS endpoint.  However rather than using a well known configuration for an Aptos project, it makes more sense to use a project ID owned by Google.  If the owner were to delete the project, then the well known endpoint would also disappearing stalling further updates to the JWK set.
+
+## Alternative solutions
+
+The alternative is to support these accounts via [Federated Keyless](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-96.md).  
+
+This comes with drawbacks as now every Keyless integration with Firebase needs to manage an account to update the JWK set.  This also introduces another point of variance when it comes to address derivation and it is needed to remember the address of the account chosen by the dApp developer to get the correct address.  Given that the JWKs update across all Firebase based Federated Keyless Accounts, it makes more sense to leverage JWK consensus[^aip-67].
```

Comment:
isn't address just derived from (iss, aud, uid, pepper)?
who will have issue with the derivation?

Reply:
Also not clear to me. Is the idea that address derivation varies more integrating federated keyless with firebase?